cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/apply/ApplicationRequestsPlaner.java

```diff
@@ -1,164 +1,273 @@
 import cloud.foundry.cli.logic.diff.change.CfChange;
+import cloud.foundry.cli.logic.diff.change.ChangeType;
+import cloud.foundry.cli.logic.diff.change.container.CfContainerChange;
+import cloud.foundry.cli.logic.diff.change.container.CfContainerValueChanged;
+import cloud.foundry.cli.logic.diff.change.map.CfMapChange;
+import cloud.foundry.cli.logic.diff.change.map.CfMapValueChanged;
 import cloud.foundry.cli.logic.diff.change.object.CfNewObject;
-import cloud.foundry.cli.logic.diff.change.object.CfObjectValueChanged;
 import cloud.foundry.cli.logic.diff.change.object.CfRemovedObject;
 import cloud.foundry.cli.operations.ApplicationsOperations;
+import org.cloudfoundry.operations.applications.ApplicationHealthCheck;
+import org.reactivestreams.Publisher;
 import reactor.core.publisher.Flux;
-
-import java.util.Arrays;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
+import reactor.core.publisher.Mono;
 
 import javax.annotation.Nonnull;
+import java.util.*;
+import java.util.function.Predicate;
 
 /**
  * This class is responsible to build the requests in the context of
  * applications according to the CfChanges. The class does create the request
  * tasks by implementing the {@link CfChangeVisitor} interface.
  */
-public class ApplicationRequestsPlaner extends RequestsPlaner {
+public class ApplicationRequestsPlaner {
 
     private static final Log log = Log.getLog(ApplicationRequestsPlaner.class);
-    private static final Set<String> FIELDS_REQUIRE_RESTART = new HashSet<>(Arrays.asList("meta", "path"));
-
-    private final ApplicationsOperations appOperations;
-    private final String applicationName;
 
-    private ApplicationRequestsPlaner(ApplicationsOperations appOperations, String applicationName) {
-        this.appOperations = appOperations;
-        this.applicationName = applicationName;
-    }
+    private static final Map<String, Field> FIELDS_REQUIRE_RESTART = new HashMap<String, Field>() {{
+        put("meta", new ScalarField(ApplicationBean.class, "meta", String.class));
+        put("path", new ScalarField(ApplicationBean.class, "path", String.class));
+        put("buildpack", new ScalarField(ApplicationManifestBean.class, "buildpack", String.class));
+        put("command", new ScalarField(ApplicationManifestBean.class, "command", String.class));
+        put("stack", new ScalarField(ApplicationManifestBean.class, "stack", String.class));
+        put("healthCheckType", new ScalarField(ApplicationManifestBean.class,
+                "healthCheckType",
+                ApplicationHealthCheck.class));
+        put("healthCheckHttpEndpoint",
+                new ScalarField(ApplicationManifestBean.class,
+                "healthCheckHttpEndpoint",
+                        String.class));
+        put("memory", new ScalarField(ApplicationManifestBean.class, "memory", Integer.class));
+        put("disk", new ScalarField(ApplicationManifestBean.class, "disk", Integer.class));
```

Comment:
I would prefer to store the fieldnames into constant string fields. That way we don't have them twice in every `put`-call (which could prevent typing errors and ease field-renamings)